src/chapters/angular-lsn3/exercises.rst

```diff
@@ -11,31 +12,255 @@ The starter code for the exercises is in the same
+button.
+
+.. sourcecode:: html+ng2
+      
+      <button (click) = "handleTakeOff()">Take Off</button>
+
+Back in ``app.component.ts``, we'll define this listener.
+The confirm method will look the same as before, but this
+time we can use a few less lines of code to update the view.
+
+.. sourcecode:: TypeScript
+   :linenos:
+
+   handleTakeOff() {
+      let result = window.confirm('Are you sure the shuttle is ready for takeoff?');
+      if (result) {
+         this.status = this.shuttleTakeOff;
+      }
+   }
+
+Follow the same pattern to handle the *Land* and *Abort Mission*
+click events.
+
+Movement Buttons
+^^^^^^^^^^^^^^^^
+
+Next, we'll tackle the ``Up``, ``Down``, ``Left``, and ``Right`` buttons that
+move the rocket. The ``movement`` formula is the same as we've used before:
+
+.. sourcecode:: TypeScript
+   :linenos:
+
+   let movement = parseInt(img.style.left) + 10 + 'px';
 
-Maybe change text color if fuel level drops too low?
 
-Maybe enable/disable buttons based on whether ``cargoHold`` is full?
+But now, instead of using the ``getElementById`` method, we'll
+access the ``img`` element by passing it in to the click
+event.
 
-Enable ``Launch`` button when all checks pass?
+In ``app.component.html``, label the ``img`` element so we can reference it:
 
-Fly the Rocket?
+.. sourcecode:: html+ng2
+      
+      <img #rocketImage src="assets/images/LaunchCode_rocketline_white.png" height = "75" width = "75" [style.left]="0" [style.bottom]="0"/>
+
+While you're here, add the click handler to the *Right* button:
+
+.. sourcecode:: html+ng2
+      
+      <button (click)="handleRightClick(rocketImage)">Right</button>
+
+Now in ``app.component.ts`` we can write the ``handleRightClick()``:
+
+.. sourcecode:: TypeScript
+   :linenos:
+
+   handleRightClick(rocketImage) {
+      let movement = parseInt(rocketImage.style.left) + 10 + 'px';
+      rocketImage.style.left = movement;
+   }
+
+Follow the same pattern for the other directional buttons, modifying the
+movement formula as needed. For the *Up* and *Down* buttons, we'll want
+to also update the current status object's height property.
+
+For example, the *Down* click handler could look like this:
+
+.. sourcecode:: TypeScript
+   :linenos:
+
+   handleDownClick(rocketImage) {
+    let movement = parseInt(rocketImage.style.bottom) - 10 + 'px';
+    rocketImage.style.bottom = movement;
+    this.status.height = this.status.height - 10000;
+   }
+
+Update the Control Button Click Handlers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Along those same lines, we'll want to modify a couple of our control 
+button handlers to update ``rocketImage``'s position when the status
+changes. Pass in ``rocketImage`` to your *Land* and *Abort Mission* 
+handlers and add the following:
+
+.. sourcecode:: TypeScript
+
+   rocketImage.style.bottom = '0px';
+
+New Requirements
 ----------------
+
+You'll notice, a user can move the rocket before the shuttle is officially taken off. One can also
```

Comment:
Rephrasing suggestion: How about, "Right now, a user can move the rocket before it has officially taken off or abort the mission while the rocket is still on the ground."